core/src/main/java/com/linecorp/armeria/common/HttpHeadersBase.java

```diff
@@ -294,19 +431,56 @@ final void status(int statusCode) {
 
     final void status(HttpStatus status) {
         requireNonNull(status, "status");
-        set(HttpHeaderNames.STATUS, status.codeAsText());
+        cache.put(HttpHeaderNames.STATUS, status);
+        set(HttpHeaderNames.STATUS, status.codeAsText(), false);
+    }
+
+    final void contentLength(long contentLength) {
+        checkArgument(contentLength >= 0, "contentLength: %s (expected: >= 0)", contentLength);
+        cache.put(HttpHeaderNames.CONTENT_LENGTH, contentLength);
+        final String contentLengthString;
+        if (contentLength <= 1000) {
+            contentLengthString = StringUtil.toString((int) contentLength);
+        } else {
+            contentLengthString = Long.toString(contentLength);
+        }
```

Comment:
nit: How about adding `toString` method that takes a long as an argument in `StringUtil`?
```java
public static String toString(long num) {...}
```